### PR TITLE
db-edu.pacha.dev as fallback

### PR DESCRIPTION
**Pull request by pachamaltese**
this is the most "harmless" way of adding a 2nd SQL database as fallback without touching much of the remaining code

### CODE REVIEW

File: R/financial.R [hunk 1]
```diff
@@ -17,14 +17,30 @@
...
+      DBI::dbConnect(
+        RMariaDB::MariaDB(),
+        username = "student",
+        password = "tx5mvyRQqD",
+        dbname = "loan_application",
+        host = "db-edu.pacha.dev"
+      ),
+      silent = TRUE
+    )
+  }
```
Comment (krlmlr) [inline R/financial.R:42]: Can you extract a function here, please? The function should have an argument `source = "relational.fit"`, the other option would be `source = "db-edu"` .

File: R/financial.R [hunk 2]
```diff
@@ -17,14 +17,30 @@
 dm_financial <- function() {
   stopifnot(rlang::is_installed("RMariaDB"))
 
-  my_db <- DBI::dbConnect(
-    RMariaDB::MariaDB(),
-    username = "guest",
-    password = "relational",
-    dbname = "Financial_ijs",
-    host = "relational.fit.cvut.cz"
+  my_db <- try(
```
Comment (krlmlr) [inline R/financial.R:20]: `tryCatch()` is a little safer for programming.

**Review by krlmlr — COMMENTED** [review]
Thanks for looking into it!

File: dm.Rproj [hunk 3]
```diff
@@ -18,4 +18,4 @@ StripTrailingWhitespace: Yes
 BuildType: Package
 PackageUseDevtools: Yes
 PackageInstallArgs: --no-multiarch --with-keep.source
-PackageRoxygenize: rd,collate,namespace
+PackageRoxygenize: rd,collate,namespace,vignette
```
Comment (krlmlr) [inline dm.Rproj:21]: Why do we need this?

File: R/financial-db-con.R [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      username = "student",
+      password = "tx5mvyRQqD",
+      dbname = "loan_application",
+      host = "db-edu.pacha.dev"
+    )
+  }
+
+  if (source == "relational.fit") {
+    con <- tryCatch(relational_con(), silent = TRUE)
+    if (inherits(my_db, "try-error")) {
```
Comment (krlmlr) [inline R/financial-db-con.R:35]:
```suggestion
    con <- tryCatch(relational_con(), error = identity)
    if (inherits(my_db, "error")) {
```

File: R/financial-db-con.R [hunk 5]
```diff
@@ -0,0 +1,43 @@
+#' Connection to SQL Financial Database
+#' @description Connects to relational.fit.cvut.cz unless the service is
+#' unavailable, in which case db-edu.pacha.dev is used as a fallback
+#' @param source (type: character) which service to use, by default is
+#' "relational.fit" but it can be set to "db-edu" as well
+#' @return A `MariaDBConnection` object
+#' @export
```
Comment (krlmlr) [inline R/financial-db-con.R:7]: Let's keep this private.

```suggestion
```

File: R/financial.R [hunk 6]
```diff
@@ -17,13 +17,7 @@
 dm_financial <- function() {
```
Comment (krlmlr) [inline R/financial.R:17]: Let's expose `source` here:

```suggestion
dm_financial <- function(source = "relational.fit") {
```

**Review by krlmlr — COMMENTED** [review]
Thanks, looks good!

File: dm.Rproj (hunk 3 above)
Comment (pachamaltese) [reply dm.Rproj:21]: no need ;)

**Comment by krlmlr** [comment]
Thanks. I tweaked and merged.

**Comment by pachamaltese** [comment]
let's talk during the rest of the week

**Pull request merged**